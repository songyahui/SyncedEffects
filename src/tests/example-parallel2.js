"use hopscript"

var hh = require("hiphop");

var prg = <hh.module J>
  <hh.local I>
    <hh.Parallel>
      <hh.if I>
	<hh.emit J/>
      </hh.if>
      <hh.emit I/>
    </hh.Parallel>
  </hh.local>
</hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "parallel2");
