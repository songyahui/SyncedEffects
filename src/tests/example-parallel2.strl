module parallel2:

output J;

signal I in
       present I then emit J end || emit I
end signal

end module