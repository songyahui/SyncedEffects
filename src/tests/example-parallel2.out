parallel2> ;
--- Output: J
parallel2> ;
--- Output:
parallel2> ;
--- Output:
